packages/kbn-check-saved-objects-cli/src/migrations/removed_types/detect_removed_types.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { MigrationSnapshot } from '../../types';
+
+/**
+ * Detects for new removed types by comparing two snapshots and identifying
+ * types that exist in 'from' but not in 'to' and not already in removed_types.json
+ */
+export function detectRemovedTypes(
```

Comment:
NIT maybe we can update the name to something like `detectNewRemovedTypes(...)` to make it more clear that we're only returning those that we were not accounting for already.